pkgmgmtutils/internal/formula_bytes_updater_test.go

```diff
@@ -54,4 +56,46 @@ var _ = Describe("FormulaBytesUpdater", func() {
 			)
 		}).To(Panic())
 	})
+
+	It("can replace homebrew-core version and sha", func() {
+		testVersion := "v7.8.9"
+		testVersionSha := "gitcommitsha"
+		formula := formula_updater_types.FormulaOptions{
+				Name:            "homebrew-core/glooctl",
+				FormulaName:     "glooctl",
+				Path:            "Formula/glooctl.rb",
+				RepoOwner:       "solo-io",
+				RepoName:        "homebrew-core",
+				PRRepoOwner:     "homebrew",
+				PRRepoName:      "homebrew-core",
+				PRBranch:        "master",
+				PRDescription:   "Created by Solo-io Bot",
+				PRCommitName:    "Solo-io Bot",
+				PRCommitEmail:   "bot@solo.io",
+				VersionRegex:    `tag:\s*"v([0-9.]+)",`,
+				VersionShaRegex: `revision:\s*"(.*)"`,
+		}
+
+		// GitHub API docs: https://developer.github.com/v3/repos/contents/#get-contents
+		gclient := github.NewClient(nil)
+		ctx := context.Background()
+		fileContent, _, _, err := gclient.Repositories.GetContents(ctx, formula.RepoOwner, formula.RepoName, formula.Path, &github.RepositoryContentGetOptions{
+			Ref: "refs/heads/master",
+		})
+		Expect(err).To(Not(HaveOccurred()))
+
+		c, err := fileContent.GetContent()
+		Expect(err).To(Not(HaveOccurred()))
+
+		b, err := internal.UpdateFormulaBytes(
+			[]byte(c),
+			"7.8.9",
+			"gitcommitsha",
+			&formula_updater_types.PerPlatformSha256{},
+			&formula_updater_types.FormulaOptions{VersionRegex: formula.VersionRegex, VersionShaRegex: formula.VersionShaRegex},
+		)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(b)).To(ContainSubstring(testVersion))
```

Comment:
Aren't we replacing the version that matches the regex with `7.8.9`? Why aren't we comparing that it matches our replaced value?

Reply:
Yes, the GetContents gets the glooctl homebrew-core contents, and then that version is replaced with our testVersion `v7.8.9`. So the original value is not `v7.8.9`, but it gets set to that by UpdateFormulaBytes